evm/src/generation/prover_input.rs

```diff
@@ -411,30 +407,30 @@ fn get_proofs_and_jumpdests(
     const PUSH32_OPCODE: u8 = 0x7f;
     let (proofs, _) = CodeIterator::until(code, largest_address + 1).fold(
         (vec![], 0),
-        |(mut proofs, acc), (pos, opcode)| {
-            let has_prefix = if let Some(prefix_start) = pos.checked_sub(32) {
-                code[prefix_start..pos]
-                    .iter()
-                    .enumerate()
-                    .fold(true, |acc, (prefix_pos, &byte)| {
+        |(mut proofs, last_proof), (addr, opcode)| {
+            let has_prefix = if let Some(prefix_start) = addr.checked_sub(32) {
+                code[prefix_start..addr].iter().enumerate().fold(
```

Comment:
In the case that the predicate returns `false`, then `acc` will be `false` in the next iteration, but once it's set to `false`, it can't ever flip back to `true` I think – is that what you want here? If that is correct, then why not use [std::iterator::any](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) here to short-circuit the iteration at the first `false`?

Reply:
> but once it's set to false, it can't ever flip back to true

Yes, because you're computing a big and over 32 boolean vars. 

>  is that what you want here? If that is correct, then why not use [std::iterator::any](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.any) 

Sure, that looks much better. Though you would need `iterator::all` 

Thanks!